src/commands/delete.ts

```diff
@@ -52,19 +53,32 @@ function getItemToDelete(contextItem: TektonNode, selectedItems: TektonNode[]):
 }
 
 async function doDelete(items: TektonNode[], toRefresh: Refreshable, commandId?: string): Promise<void | string> {
+  const taskRunList = await getTaskRunResourceList();
```

Comment:
A couple of comments here.
1) Wouldn't it be better to put this call after checking the setting is set to true? If the setting is false this value will never be used and you executed a command for nothing
2) Why do you use taskruns? Taskruns are immutable object that describe how a task has been run but this doesn't tell you if the task is actually used somewhere. Maybe it was only used in the past.
E.g i have a pipeline A that uses the task B and i execute it (this does create pipelinerun/taskrun), now i change the pipeline A to use the task C. If i want to delete the task B, i will be shown the message that the task is used on the cluster even if this is not actually true.